update content for Contributing.md (#43478)
joomla/joomla-cms: .github/CONTRIBUTING.md

2) Follow the [Joomla! Coding Standards](https://developer.joomla.org/coding-standards.html).

3) When filing an issue or opening a PR, please include a clear title and description. The title should be a short summary of an issue and, if possible, should include a reference to an open issue. For example, `Invalid Query in com_admin (Ref #1234)` would be sufficient. All issues and PRs should include a description with as much detail as possible.
   If it is a PR, include what the issue is, what the PR is addressing, testing instructions and environmental information (PHP version, database driver and version, and other data you can retrieve from your site's system information view) in case the issue is specific to certain environments. If additional information is needed, please be prepared to provide it as our community members review your submission.

4) Report security issues to the Joomla! Security Strike Team (JSST) at security@joomla.org or use the [JSST contact form](https://developer.joomla.org/contact-security-team.html). Please do not use the public tracker for security issues.

> 5) If you have a feature request open a discussion to define the feature request and discuss possible solutions. Discussions can be coverted into issues when the request is so clear that a developer can start codig on the feature. In this process you will get feedback from the maintainers, if the feature is something for the Joomla core distribution.

Reviewer: ```suggestion
5) If you have a feature request, then please open a discussion to define the feature request and discuss possible solutions. Discussions can be converted into issues when the request is defined sufficiently that a developer can start coding the feature. In this process you will get feedback from the maintainers, if the feature is something for the Joomla core distribution or not.
```

Reviewer 2: typo 5) ... Discussions can be coverted -> converted

Please be patient as not all items will be tested immediately (remember, all bug testing for the Joomla! CMS is done by volunteers) and be receptive to feedback about your code.

#### Branches

> As long as 4.4 is in maintaince mode, what means that we still fixing bugs, PRs should be made to the `4.4-dev` branch. Merged bugfixes will be upmerged into the current 5.x branch. If a bug is only in the 5.x series the PR should be made to the current 5.x branch (currently 5.1).

Reviewer: ```suggestion
While 4.4 is in maintenance mode, ie we are still fixing bugs, PRs should be made to the `4.4-dev` branch. Merged bugfixes will be upmerged into the current 5.x branch. If a bug is only in the 5.x series the PR should be made to the current 5.x branch (currently 5.1).
```

| Branch | Purpose |
| ------ | ------- |

> | 4.4-dev | Branch for the current 4.x Joomla version. Currently in maintaince mode |

Reviewer: ```suggestion
| 4.4-dev | Branch for the current 4.x Joomla version. Currently in maintenance mode |
```
